### ISSUE TITLE
[Feature] Two-way Sync

### ISSUE THREAD

**Issue by AndyXheli** [nextcloud/ios#2670]
Default one-way sync: assets sync from Photos to Server only.
One-way smart sync: assets deleted from Server aren't re-synced from cloud.
Two-way sync: assets deleted in Photos are deleted from server . Assets deleted from server are deleted in Photos.
I would at least ask for the one-way smart sync; without that

**Comment by ne20002** [comment]
It's basically two different features.
* the auto upload is to save your photos to the server, regardless of what you do with/on the phone. E.g to save the photos to then delete it on the device to save space.
* a sync is what icloud photos offer. But I think the api is not existing that an app can implement similar behaviour with the photo roll.

**Comment by AndyXheli** [comment]
Check out https://github.com/immich-app/ their app syncs with the ios photos app and the server and tells the users what on the server and whats on the iphone and you can mark it as download and it will show on the iphone photos. It actually a really cool product ive been testing it. I think Nextcloud should looking into something similar to Immich.

**Comment by ne20002** [comment]
I'll give it try. As far as I know, an app can't remove files from photo roll without a popup confirming this action (which prevents a sync in background).

If immich is offering the requested functionality, why should Nextcloud implement this as well? Seems to be unnecessarily duplicated functionality.

**Comment by ne20002** [comment]
There I have different opinion. Basic pattern: separation of concerns.
I would go for the current upload with the Nextcloud app and for photo albums syncing with immich or some other app and a Nextcloud server app as backend for those.

**Comment by AndyXheli** [comment]
But Nextcloud client dose not detect duplicate Photos properly. 

Example: 

User download file from Shared Photos into iphone > auto upload is on Nextcloud client detect new phone or video and reuploads it into default Photos folder. Now the users need to delete all the photos that where reuploaded but already exist in Nextcloud Under Shared Photos Folder. This is not ideal for users and it dose not make sense as to why the files keeps getting reuploaded when the file already exist on the server. 

There has to be a better way to sync photos then this.

**Comment by ne20002** [comment]
I believe the original intention of the Nextcloud auto upload was work as a upload/backup solution so a upload is triggered on every add to the photo roll. The user may then delete photos on the device to save space.
Syncing is a completely different use case. I would leave this to other app with other backends including e.g. albums and tagging.
I doubt both features can work together side by side so if a user wants syncing, he should disable auto upload of photo roll.

**Comment by zeratax** [comment]
a proper sync feature is exactly what’s needed to replace icloud. sure you could probably set up this immich project to do that. but that requires access to the nextcloud server, which many nextcloud users don’t have. this is obviously desirable functionality for a cloud storage app, since i can already sync any other folder and this is also possible in the android app.

**Comment by AndyXheli** [comment]
@zeratax i would agree with you!! The hope is for nextcloud to support this feature

**Comment by AndyXheli** [comment]
Thanks for the update, @marinofaggiana!

I’m looking for a way to address duplicate photos between my iPhone and the Nextcloud server. Specifically, I want to avoid having the Nextcloud client re-upload photos that are already on the server.

Here’s the scenario: I have a photo stored in a folder on Nextcloud. I save that photo to my iPhone or iPad. When the Nextcloud client detects this new photo on my device, it re-uploads it to the server, leading to duplicates.

Is there a method to implement a sync that compares data between the iPhone and the Nextcloud server to eliminate these duplicates? I'm not necessarily looking for a full two-way sync but rather a solution that prevents this duplication issue.

Does this make sense? If you have any suggestions or solutions, I’d appreciate it!

**Comment by AndyXheli** [comment]
@marinofaggiana Yes, via code. 

I was looking at that immich-app and the Synology photos client and that how it seems they are doing it and i wonder what they are doing in the code

**Comment by AndyXheli** [comment]
Hmm im not sure.

 So For example named on Photos IMG_0621 and it has been uploaded on my nextcloud as 24-08-17 09-30-43 0621.jpg. So I selected Save Live Photo to Photo Album on the nextcloud client and it saved it back down to my iphone and it has a new file name and now i see two of the same photos under my iphone and now i also see that the photo has been reuploaded as  24-08-17 09-30-43 73BB.jpg under nextcloud

![image](https://github.com/user-attachments/assets/c875eaf1-660c-463d-924b-4eab5b98fb0d)

**Comment by AndyXheli** [comment]
Oh i see what your talking about! Yea i think that would be cool if that can be done

**Comment by AndyXheli** [comment]
hmm what about if users have photos outside of the "Photos" folder lets say they also have another folder called Shared Photos. how would that work or can there be a  metadata search or when the client i setup the photos tab its already looking that all the photos and folders so would there be a way to cache that locally and nextcloud client check local cache or would that be a nightmare ?

**Comment by AndyXheli** [comment]
Ahhh okay! Hmmmm what do you think the best option would be then? or do you want to think about it ?
